File: todo/src/main/java/com/king/todo/inject/module/AppModule.java
package com.king.todo.inject.module;

import android.content.Context;
import android.content.res.Resources;

import com.king.todo.inject.qualifier.context.AppContext;
import com.king.todo.inject.scope.ApplicationScope;
import com.king.todo.ui.MoneyApplication;

import dagger.Module;
import dagger.Provides;

/**
 * project：cutv_ningbo
 * description：
 * create developer： admin
 * create time：10:10
 * modify developer：  admin
 * modify time：10:10
 * modify remark：
 *
 * @version 2.0
 */
@Module
public class AppModule {
    private final MoneyApplication app;
    public AppModule(MoneyApplication app) {
        this.app = app;
    }


    @AppContext
    @ApplicationScope
    @Provides
    Context getApplicationContext() {
        return app;
    }

    @Provides
    @ApplicationScope
    Resources provideResources() {
        return app.getResources();
    }
}

File: todokotlin/src/main/java/com/union/bangbang/todokotlin/dagger/module/AppModule.kt
package com.union.bangbang.todokotlin.dagger.module

import android.app.Application
import android.content.Context
import android.content.res.Resources
import com.union.bangbang.todokotlin.TodoApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
不乱于心，不困于情。不畏将来，不念过往。如此，安好!

深谋若谷，深交若水。深明大义，深悉小节。已然，静舒!

善宽以怀，善感以恩。善博以浪，善精以业。这般，最佳!

勿感于时，勿伤于怀。勿耽美色，勿沉虚妄。从今，进取!

无愧于天，无愧于地。无怍于人，无惧于鬼。这样，人生!
 */
@Module
class AppModule constructor() {
    @Provides
    fun getApplicationContext(application: Application): Context {
        return application
    }

    @Provides
    fun provideResources(context: Context): Resources {
        return context.getResources()
    }
}